modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/action/DiscardChangesMVCActionCommand.java

```diff
@@ -42,8 +45,24 @@ protected void doProcessAction(
 			actionRequest, "modelClassNameId");
 		long modelClassPK = ParamUtil.getLong(actionRequest, "modelClassPK");
 
-		_ctCollectionService.discardCTEntry(
-			ctCollectionId, modelClassNameId, modelClassPK);
+		if ((modelClassNameId <= 0) || (modelClassPK <= 0)) {
+			long[] ctEntryIds = StringUtil.split(
+				ParamUtil.getString(actionRequest, "ctEntryIds"), 0L);
+
+			for (long ctEntryId : ctEntryIds) {
+				CTEntry ctEntry = _ctEntryLocalService.fetchCTEntry(ctEntryId);
```

Comment:
can you move this into the service so it can all happen in a single transaction

Reply:
or you can create a transaction in the action.  look at BaseTransactionalMVCActionCommand

Reply:
So just change `doProcessAction` to `doTransactionalCommand`, right?